nat.go

```diff
@@ -34,20 +37,90 @@ type NAT interface {
 	DeletePortMapping(protocol string, internalPort int) (err error)
 }
 
+// DiscoverNATs returns all NATs discovered in the network.
+func DiscoverNATs(ctx context.Context) <-chan NAT {
+	nats := make(chan NAT)
+
+	go func() {
+		defer close(nats)
+
+		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Just curious what is the difference from the first input ctx argument?

Reply:
That's a bug.